### DISCUSSION TITLE
Rules for pull request?

### DISCUSSION THREAD

**Discussion by ip6li** [Alovoa/alovoa#293] [Show and tell]
Are there any rules regarding code quality, tests etc. for pull requests?

**Comment by Nonononoki** [comment]
Minimum: Should be formatted correctly, Java could should have Junit tests.

**Reply by ip6li** [reply]
Thank you for fast reply. Regarding unit tests I have to implement TestContainers to test my new DataSource class with Hashicorp Vault. Test must configure Hashicorp Vault by script to provide MySQL credential rotation.
